addons/website_blog/models/website_blog.py

```diff
@@ -219,49 +188,45 @@ def _check_for_publication(self, cr, uid, ids, vals, context=None):
                         'blog_slug': slug(post.blog_id),
                         'post_slug': slug(post),
                     },
-                    subtype='website_blog.mt_blog_blog_published',
-                    context=context)
+                    subtype='website_blog.mt_blog_blog_published')
             return True
         return False
 
-    def create(self, cr, uid, vals, context=None):
-        if context is None:
-            context = {}
+    @api.model
+    def create(self, vals):
         if 'content' in vals:
-            vals['content'] = self._postproces_content(cr, uid, None, vals['content'], context=context)
-        create_context = dict(context, mail_create_nolog=True)
-        post_id = super(BlogPost, self).create(cr, uid, vals, context=create_context)
-        self._check_for_publication(cr, uid, [post_id], vals, context=context)
-        return post_id
+            vals['content'] = self._postprocess_content(None, vals['content'])
+        post = super(BlogPost, self.with_context(mail_create_nolog=True)).create(vals)
+        post._check_for_publication(vals)
+        return post
 
-    def write(self, cr, uid, ids, vals, context=None):
+    @api.multi
+    def write(self, vals):
         if 'content' in vals:
-            vals['content'] = self._postproces_content(cr, uid, ids[0], vals['content'], context=context)
-        result = super(BlogPost, self).write(cr, uid, ids, vals, context)
-        self._check_for_publication(cr, uid, ids, vals, context=context)
+            vals['content'] = self._postprocess_content(self.id, vals['content'])
```

Comment:
how can you get self.id if @api.multi ? Use directly  self._postprocess_content(vals['content'])?